### PR TITLE
observability: remove outdated prometheus page, move to monitoring

### PR DESCRIPTION
**Pull request by bobheadxi**
I did a search for Prometheus stuff and came across this very outdated page - I've moved relevant bits into the new documentation, and removed the page and references to it completely 

Also cleaned up some headers to make them easier to reference

### CODE REVIEW

**Comment by bobheadxi** [comment]
can't seem to find the broken link the check is referring to 🤔

**Comment by pecigonzalo** [comment]
I tried, but could not find it either, I only found the following links.
```
❯ rg '.*(\(.+\)).*' -r '$1' distribution/observability/monitoring_architecture.md | rg -v 'http' | rg '(\.md)'
(../../observability/monitoring.md)
(../../observability/monitoring_pillars.md#long-term-vision)
(../../observability/monitoring_pillars.md)
(../../observability/monitoring.md#adding-monitoring)
(../../observability/monitoring_pillars.md#long-term-vision))
(../../observability/monitoring_pillars.md#long-term-vision)
(../../observability/monitoring_pillars.md#long-term-vision)
```

**Comment by bobheadxi** [comment]
@pecigonzalo I updated the PR based on #1785 (which IMO moves the architecture to a better place) and it's good to go now 👌

**Review by daxmc99 — APPROVED** [review]

File: handbook/engineering/observability/prometheus.md [hunk 1]
```diff
@@ -1,130 +0,0 @@
...
-curl -X POST http://localhost:9090/-/reload
-```
-
-to your running Prometheus in your dev Sourcegraph instance and it will reload the configs with your changes.
-
-## Find available metrics
-
-If you need to find where metrics are declared or updated you can use Sourcegraph itself to search if you have a metric name. Sometimes
-the metrics are hard to find because their name declarations are not literal strings but are concatenated instead in code from variables.
-You can try a specialized tool called `promgrep` to find them. Run the tool in the root `sourcegraph/sourcegraph` source directory.
```
Comment (daxmc99) [inline handbook/engineering/observability/prometheus.md:62]: TIL about `promgrep`

**Pull request merged**